_drafts/wsl2-firmware-development.md

```diff
@@ -0,0 +1,655 @@
+- BLE, and probably other kernel modules that I haven’t thought of, are not enabled in the default WSL2 configuration.
+- Performance is not as good as running Linux on bare metal.
+
+### Conclusion
+
+Ultimately, choosing a development environment depends on what helps you to
+provide the most value to your clients.  Decisions about security, performance,
+features, and stability must be weighed distinctly within each organization or
+project.  Personally, I’ve found a nice balance by using WSL2 on Windows 11. 
+I get all of the security, stability, compatibility, and support of Windows while
+being able to use my preferred tools in a Linux environment.  My primary concern,
+performance, has been eased by quantifying that a WSL2 Zephyr build is only ~12% slower
+than bare metal Linux.  If you compile big Zephyr projects with 40s build times - 
+_pristine instead of incremental_ - 20 times a day (I don’t), WSL2 costs you an extra 1m 30s
+of compile time.  But what does maintaining desktop Linux cost each day?  How
+much time do you have for fixing ALSA, compiling a tool that didn’t have a binary
+release, fiddling with WINE, or setting up a Windows VM?
+
+If you’re interested in trying out WSL2 for firmware development, continue reading below for a complete tutorial.  This is a living tutorial document maintained by Intercreate and we would welcome feedback and contributions at [GitHub](https://github.com/intercreate/docs/blob/main/Windows/wsl2.md)!
+Depending on the state of your Windows host, setting up WSL2 for firmware
+development should take between 10 and 30 minutes.
+
+## Tutorial
+
+This guide provides instructions for setting up an environment for developing,
+debugging, and programming embedded systems firmware in the Windows Subsystem
+for Linux (WSL2).
+
+Contributions by Michael Brust, Alden Haase, JP Hutchins, and Ishani Raha.
+
+### Windows Dependencies and Configuration
+
+Each task in this section is run from within the Windows 11 environment.
+
+#### Windows 11
+
+While many of these same techniques are possible in Windows 10, WSL2 in Windows
+10 is no longer receiving important feature updates. It is recommended to run
+Windows Update until up to date.
+
+[Installation Documentation](https://support.microsoft.com/en-us/windows/ways-to-install-windows-11-e0edbbfb-cfc5-4011-868b-2ce77ac7c70e)
+
+#### Windows Terminal
+
+Windows Terminal is a GPU accelerated terminal host that simplifies access to
+the Windows shells, e.g. Command Prompt, PowerShell, PowerShell Core, and WSL2.
+
+Windows Terminal should already be installed on recent versions of Windows 11.
+In the Start Menu, begin typing `windows terminal` and select the application
+if it is found.
+
+If you don't have Windows Terminal yet, you'll need to use PowerShell to install
+it.  After this, Windows Terminal is your interface to PowerShell.
+
+```powershell
+winget install --id Microsoft.WindowsTerminal -e
+```
+
+##### Configuration
```

Comment:
can one use a terminal emulator from the WSL instance and displays its windows from within Windows? e.g. kitty, alacritty, wezterm. Not sure if the W11 terminal is required if native Linux apps are displayed in the DE.

Reply:
You can launch the terminal emulator from within WSL2 but you'd be better off launching the terminal emulator from Windows and SSH to WSL2.  WSL2 has no desktop environment.

Reply:
I think that WSL2 natively uses their WSLg thingy now: https://github.com/microsoft/wslg?tab=readme-ov-file#wslg-architecture-overview

But as far as i can tell performance is not better or worse than X forwarding.  I'd rather render graphics natively.